### PR TITLE
subsystem-bench: adjust test config to Kusama

### PR DESCRIPTION
**Pull request by AndreiEres**
Fixes https://github.com/paritytech/polkadot-sdk/issues/3528

```rust
latency:
    mean_latency_ms = 30 // common sense
    std_dev = 2.0 // common sense
n_validators = 300 // max number of validators, from chain config
n_cores = 60 // 300/5
max_validators_per_core = 5 // default
min_pov_size = 5120 // max
max_pov_size = 5120 // max
peer_bandwidth = 44040192 // from the Parity's kusama validators
bandwidth = 44040192 // from the Parity's kusama validators
connectivity = 90 // we need to be connected to 90-95% of peers
```

### CODE REVIEW

**Comment by AndreiEres** [comment]
image
https://grafana.teleport.parity.io/goto/HQdBDO0Ig?orgId=1

We could've used these values assuming 90 and 99 percentiles
```rust
min_pov_size = 64 // 108 KiB, 90%
max_pov_size = 2024 // 1.78 MiB, 99%
```

**Comment by AndreiEres** [comment]
Based on kusama telemetry:
- average bandwidth = 1555613
- median bandwidth = 752370

I'd use a value close to median:
```rust
peer_bandwidth = 819200 // median bandwidth is 752370
bandwidth = 819200 // median bandwidth is 752370
```

**Comment by AndreiEres** [comment]
> Based on kusama telemetry:
> 
>     * average bandwidth = 1555613
> 
>     * median bandwidth = 752370

Another (and better) option is to use the reference hw recommendation: [65536000](https://wiki.polkadot.network/docs/maintain-guides-how-to-validate-polkadot#reference-hardware)

File: polkadot/node/network/availability-distribution/tests/availability-distribution-regression-bench.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 //! availability-read regression tests
 //!
-//! TODO: Explain the test case after configuration adjusted to Kusama
+//! Simulated process of availability reading on Kusama
```
Comment (sandreim) [inline polkadot/node/network/availability-distribution/tests/availability-distribution-regression-bench.rs:19]: I'd go for something like `Availability read benchmark based on Kusama parameters and scale.`

File: polkadot/node/subsystem-bench/src/lib/configuration.rs [hunk 2]
```diff
@@ -35,24 +35,39 @@ pub struct PeerLatency {
...
+
+// Based on number of kusama validators producing blocks
+fn default_n_cores() -> usize {
+	50
+}
+
+// PoV size, KiB. Based on empirical data from Kusama
 fn default_pov_size() -> usize {
-	5120
+	2 * 1024
```
Comment (sandreim) [inline polkadot/node/subsystem-bench/src/lib/configuration.rs:50]: Let's use max value here such that any degradation has more impact.

File: polkadot/node/subsystem-bench/src/lib/configuration.rs [hunk 3]
```diff
@@ -35,24 +35,39 @@ pub struct PeerLatency {
...
 fn default_connectivity() -> usize {
-	100
+	90
 }
 
-// Default backing group size
+// Based on Kusama `max_validators_per_core`
 fn default_backing_group_size() -> usize {
-	5
+	3
```
Comment (sandreim) [inline polkadot/node/subsystem-bench/src/lib/configuration.rs:70]: This has to be kept in sync with the actual value from chain. Let's keep it 5 here and override in specific benchmarks.

File: polkadot/node/subsystem-bench/src/lib/configuration.rs [hunk 4]
```diff
@@ -35,24 +35,39 @@ pub struct PeerLatency {
 	pub std_dev: f64,
 }
 
-// Default PoV size in KiB.
+// Based on Kusama `max_validators`
+fn default_n_validators() -> usize {
+	300
+}
+
+// Based on number of kusama validators producing blocks
```
Comment (sandreim) [inline polkadot/node/subsystem-bench/src/lib/configuration.rs:43]: This is not related to the validators producing blocks. It's the total number of cores: cores assigned to specific paras plus on-demand (chain config value)

Comment (sandreim) [reply polkadot/node/subsystem-bench/src/lib/configuration.rs:43]: But we don't want to fetch all cores. We'd want to represent a single node effort. Roughly this number has to be set to `n_cores * needed_approvals / n_validators.`

Comment (sandreim) [reply polkadot/node/subsystem-bench/src/lib/configuration.rs:43]: `n_cores * needed_approvals / n_validators.`  only so for availability read, for write, we need to cover all cores as said above.

**Review by sandreim — APPROVED** [review]
